specification/cognitiveservices/data-plane/ComputerVision/v1.0/ComputerVision.json

```diff
@@ -0,0 +1,1464 @@
+                },
+                "height": {
+                    "type": "integer",
+                    "description": "Height measured from the top-left point of the face."
+                }
+            }
+        },
+        "FaceDescription": {
+            "type": "object",
+            "description": "An object describing a face identified in the image.",
+            "properties": {
+                "age": {
+                    "type": "integer",
+                    "description": "Possible age of the face."
+                },
+                "gender": {
+                    "type": "string",
+                    "description": "Possible gender of the face.",
+                    "enum": [
+                        "Male",
+                        "Female"
+                    ]
+                },
+                "faceRectangle": {
+                    "$ref": "#/definitions/FaceRectangle"
+                }
+            }
+        },
+        "ImageType": {
+            "type": "object",
+            "description": "An object providing possible image types and matching confidence levels.",
+            "properties": {
+                "clipArtType": {
+                    "type": "number",
+                    "description": "Confidence level that the image is a clip art."
+                },
+                "lineDrawingType": {
+                    "type": "number",
+                    "description": "Confidence level that the image is a line drawing."
+                }
+            }
+        },
+        "ColorInfo": {
+            "type": "object",
+            "description": "An object providing additional metadata describing color attributes.",
+            "properties": {
+                "dominantColorForeground": {
+                    "type": "string",
+                    "description": "Possible dominant foreground color."
+                },
+                "dominantColorBackground": {
+                    "type": "string",
+                    "description": "Possible dominant background color."
+                },
+                "dominantColors": {
+                    "type": "array",
+                    "description": "An array of possible dominant colors.",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "accentColor": {
+                    "type": "string",
+                    "description": "Possible accent color."
+                },
+                "isBWImg": {
+                    "type": "boolean",
+                    "description": "A value indicating if the image is black and white."
+                }
+            }
+        },
+        "AdultInfo": {
+            "type": "object",
+            "description": "An object describing whether the image contains adult-oriented content and/or is racy.",
+            "properties": {
```

Comment:
So, after I started playing around it occured to me that that on response models (like `AdultInfo`) if we make the properties not nullable:

``` json
            "isAdultContent": {
                    "x-nullable" : false,
                    "type": "boolean",
                    "description": "A value indicating if the image contains adult-oriented content."
                },```

then the models get generated with non-nullable fields. (which is kinda nice for the consumer)

```  c#
        [JsonProperty(PropertyName = "isAdultContent")]
        public bool IsAdultContent { get; set; }

// vs

        [JsonProperty(PropertyName = "isAdultContent")]
        public bool? IsAdultContent { get; set; }

```

What do you think @DavidLiCIG  ? 

if the server is always providing the value, that's not a bad plan, i'd think.

Reply:
Looks good. There are quite a few examples like this. I will need to write up some unit tests for as much coverage as possible to uncover more cases like this.